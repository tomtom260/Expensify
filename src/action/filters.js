export const addTextFilter = text=>({
    type:"ADD_TEXT",
    text:text.toLowerCase()
})

export const sortByAmount = ()=>({
    type:"SORT_BY",
    sortBy:"AMOUNT",
})

export const sortByDate = ()=>({
    type:"SORT_BY",
    sortBy:"DATE",
})

export const sortByAlphabeticalOrder = ()=>({
    type:"SORT_BY",
    sortBy:"ALPHABETICAL",
})

export const setStartDate = (startDate)=>{
    startDate===null?startDate=undefined:startDate=startDate.startOf('day').valueOf()
    return {
        type:"SET_START_DATE",
        startDate:startDate
    }
}

export const setEndDate = (endDate)=>{
    endDate===null?endDate=undefined:endDate=endDate.endOf('day').valueOf()
    return {
        type:"SET_END_DATE",
        endDate:endDate
    }
}

export const setStartAmount = (startAmount)=>({
    type:"SET_START_AMOUNT",
    startAmount
})

export const setEndAmount = (endAmount)=>({
    type:"SET_END_AMOUNT",
    endAmount
})

