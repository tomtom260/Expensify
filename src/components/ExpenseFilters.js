import React from 'react';
import 'react-dates/initialize';
import {DateRangePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import moment from 'moment';
import {connect} from 'react-redux'

import { addTextFilter, sortByAlphabeticalOrder, sortByAmount, 
    sortByDate,setEndDate,setStartDate, setStartAmount, setEndAmount } from '../action/filters';
    
export class ExpenseFilters extends React.Component{
    state={
        focusedInput:null
    }
    
    render(){
        return(
            <div>   
                <input
                    type="text"
                    value={this.props.filters.text}
                    onChange={e=>this.props.dispatch(addTextFilter(e.target.value))}
                />
                <select
                    value={this.props.filters.sortBy}
                    onChange={e=>{
                        switch(e.target.value){
                            default: 
                            case "DATE":return this.props.dispatch(sortByDate())
                            case "AMOUNT":return this.props.dispatch(sortByAmount())
                            case "ALPHABETICAL": return this.props.dispatch(sortByAlphabeticalOrder())
                        }
        
                    }}
                >
                    <option value="DATE">Date</option>
                    <option value="AMOUNT">Amount</option>
                    <option value="ALPHABETICAL">Alphabetical</option>
                </select>
                    <DateRangePicker
                        startDate={this.props.filters.startDate?moment(this.props.filters.startDate):null}
                        endDate={this.props.filters.endDate?moment(this.props.filters.endDate):null}
                        focusedInput={this.state.focusedInput}
                        onFocusChange={(focusedInput)=>this.setState(()=>({focusedInput}))}
                        onDatesChange={({ startDate, endDate }) =>{
                            this.props.dispatch(setStartDate(startDate))
                            this.props.dispatch(setEndDate(endDate))
                        }}
                        startDateId="your_unique_start_date_id"
                        endDateId="your_unique_end_date_id"
                        numberOfMonths={1}
                        hideKeyboardShortcutsPanel
                        isOutsideRange={()=>false}
                        showClearDates
                    />
                    <input
                        type="text"
                        value={this.props.filters.startAmount||""}
                        onChange={e=>{
                            if(e.target.value.match(/^\d*(\.)?(\d{1,2})?$/))
                                this.props.dispatch(setStartAmount(parseFloat(e.target.value,10)))
                        }}
                        placeholder="Start Amount"
                    />
                    <input
                        type="text"
                        value={this.props.filters.endAmount||""}
                        onChange={e=>{
                            if(e.target.value.match(/^\d*(\.)?(\d{1,2})?$/))
                                this.props.dispatch(setEndAmount(parseFloat(e.target.value,10)))
                        }}
                        placeholder="End Amount"
                    />
            </div>
        )
    }
}


const mapStateToProps= state =>({
    filters:state.filters
})

export default connect(mapStateToProps)(ExpenseFilters);
