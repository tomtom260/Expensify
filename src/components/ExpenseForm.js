import React from 'react';
import 'react-dates/initialize';
import {SingleDatePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import moment from 'moment';

class ExpenseForm extends React.Component{
    state={
        description:this.props.description||"",
        amount:this.props.amount||"",
        note:this.props.note||"",
        createdAt:moment(this.props.createdAt)||moment(Date.now()),
        buttonDisabled :this.props.history.location.pathname.includes('create'),
        showDatePicker:false
    }

    buttonDisabled = (description,amount,createdAt)=>{
        if(!description)
            return this.setState({buttonDisabled:true})
        
        if(!amount||!amount.match(/^\d+((\.)\d{1,2})?$/))
            return this.setState({buttonDisabled:true})
        
        if(!createdAt||typeof createdAt.valueOf()!=='number')
            return this.setState({buttonDisabled:true})

        return this.setState({buttonDisabled:false})
    }

    
    onSubmit = (e)=>{
        e.preventDefault()
        const expense = {
            amount:parseFloat(this.state.amount,10),
            description:this.state.description,
            createdAt:this.state.createdAt.valueOf(),
            note:this.state.note
        }
        this.props.onSubmit(expense,this.props.id)
        this.props.history.push('/');
    }


    render(){
        return(
            <form>
                <input 
                placeholder ="Description"
                autoFocus
                value={this.state.description}
                onChange={e=>{
                    const description=e.target.value;
                    this.setState({description})
                    this.buttonDisabled(description,this.state.amount,this.state.createdAt)
                }}
            />
                <input 
                placeholder ="Amount"
                value={this.state.amount}
                onChange={e=>{
                    const amount=e.target.value;
                    if(amount.match(/^\d*(\.)?(\d{1,2})?$/)){
                        this.setState({amount})
                        this.buttonDisabled(this.state.description,amount,this.state.createdAt)
                    }
                }}
                />

                <SingleDatePicker
                    date={this.state.createdAt}
                    onDateChange={createdAt => {this.setState({
                                createdAt:createdAt
                            })
                            this.buttonDisabled(this.state.description,this.state.amount,createdAt)
                        }
                    }                       
                    focused={this.state.showDatePicker}
                    onFocusChange={({focused})=>this.setState({showDatePicker:focused})}
                    numberOfMonths={1}
                    hideKeyboardShortcutsPanel
                    isOutsideRange={()=>false}
                    showClearDate
                    placeholder="DD/MM/YYYY"
                />
                <textarea 
                    placeholder="Note (Optional)"
                    value={this.state.note}
                    onChange={e=>{
                        const note=e.target.value
                        this.setState({note})
                    }}

                ></textarea>
                <button 
                    disabled={this.state.buttonDisabled}
                    onClick={this.onSubmit}
                >
                    {this.props.history.location.pathname.includes('create')?'Add':'Edit'} Expense
                </button>
            </form>
        )

    }
}


export default ExpenseForm