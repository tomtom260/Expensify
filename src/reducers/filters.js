import moment from 'moment'

const defaultFilters ={
    text:"",
    sortBy:"DATE",
    startDate:moment().startOf('month').valueOf(),
    endDate:moment().endOf('month').valueOf(),
    startAmount:undefined,
    endAmount:undefined,
}

export default (state=defaultFilters,action)=>{
    switch(action.type){
        case "ADD_TEXT":
            return {
                ...state,
                text:action.text
            }

        case "SORT_BY":
            return{
                ...state,
                sortBy:action.sortBy
            }

        case "SET_START_DATE":
            return{
                ...state,
                startDate:action.startDate
            }

        case "SET_END_DATE":
            return{
                ...state,
                endDate:action.endDate
            }

        case "SET_START_AMOUNT":
            return{
                ...state,
                startAmount:action.startAmount
            }

        case "SET_END_AMOUNT":
            return{
                ...state,
                endAmount:action.endAmount
            }

        default:
            return state
    }
}
