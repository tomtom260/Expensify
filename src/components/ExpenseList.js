import React from 'react';
import {Link} from 'react-router-dom'
import {connect} from 'react-redux'

import filterExpenses from '../selectors/filterExpenses'
import { removeExpense } from '../action/expenses';

export const ExpenseList =  props => (
    <div>
        {
            props.expenses.length===0?
                <p>No Expenses</p>:
            (props.expenses.map(el=>{
                return <div key={el.id}>
                    <Link to={`/edit/${el.id}`}>
                        <h2>{el.description}</h2>
                        <p>Amount: {el.amount}  createdAt: {el.createdAt} </p>
                    </Link>
                    <button onClick={()=>props.dispatch(removeExpense(el.id))}>Remove</button>
                </div>
            }))
        }
    </div>
)


const mapStatetoProps= state =>({
    expenses:filterExpenses(state),
})

export default connect(mapStatetoProps)(ExpenseList);