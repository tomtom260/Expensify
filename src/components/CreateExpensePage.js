import React from 'react'
import { connect } from 'react-redux';
import { addExpense } from '../action/expenses';
import ExpenseForm from './ExpenseForm';

const CreateExpensePage =(props)=>(
    <div>
        <h1>Create New Expense</h1>
        <ExpenseForm onSubmit={(expense)=>props.dispatch(addExpense(expense))} history={props.history}/>
    </div>
)

export default connect()(CreateExpensePage);