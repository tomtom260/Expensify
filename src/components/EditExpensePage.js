import React from 'react'
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense,removeExpense} from '../action/expenses'

export const EditExpensePage =(props)=>(
    <div>
        <h1>Edit Expense</h1>
        <ExpenseForm 
            onSubmit={(expense,id)=>props.dispatch(editExpense(id,expense))}
            {...props.expense} 
            history={props.history} />
        <button onClick={()=>props.dispatch(removeExpense(props.match.params.id))}>Remove Expense</button>    
    </div>
)

const mapPropsFromState = (state,props) =>{
    const expense = state.expenses.find((el)=>el.id===props.match.params.id)
    return{
        expense
    }
}
 
export default connect(mapPropsFromState)(EditExpensePage);