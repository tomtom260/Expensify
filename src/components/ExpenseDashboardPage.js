import React from 'react';
import ExpenseFilters from './ExpenseFilters'
import ExpenseList from './ExpenseList'
import ExpenseSummary from './ExpenseSummary';

export default ()=>(
    <div>
        <ExpenseSummary/>
        <ExpenseFilters />
        <ExpenseList  />  
    </div>
)
