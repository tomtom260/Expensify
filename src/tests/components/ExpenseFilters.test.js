import React from 'react';
import {shallow} from 'enzyme';
import moment from 'moment';

import {ExpenseFilters} from '../../components/ExpenseFilters'
import {filters,altFilters} from '../fixtures/filters';

test('should render ExpenseFilters correctly',()=>{
    const wrapper = shallow(<ExpenseFilters filters={filters}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseFilters correctly with alt data',()=>{
    const wrapper = shallow(<ExpenseFilters filters={altFilters}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should call dispatch with the correct action object when text filter is changed',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('input').at(0).simulate('change',{target:{value:"123abc"}})
    expect(dispatchSpy).toHaveBeenCalledWith({
        type:"ADD_TEXT",
        text:"123abc"
    })
})

test('should call dispatch with the correct action object when sortBy is changed',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('select').simulate('change',{target:{value:"AMOUNT"}})
    expect(dispatchSpy).toHaveBeenCalledWith({
        type:"SORT_BY",
        sortBy:"AMOUNT"
    })
})

test('should call dispatch with the correct action object when startAmount is changed',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('input').at(1).simulate('change',{target:{value:"123.55"}})
    expect(dispatchSpy).toHaveBeenCalledWith({
        type:"SET_START_AMOUNT",
        startAmount:123.55
    })
})

test('should NOT call dispatch startAmount is NOT valid',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('input').at(1).simulate('change',{target:{value:"123.555"}})
    expect(dispatchSpy).toHaveBeenCalledTimes(0)
})
test('should call dispatch with the correct action object when endAmount is changed',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('input').at(2).simulate('change',{target:{value:"123.55"}})
    expect(dispatchSpy).toHaveBeenCalledWith({
        type:"SET_END_AMOUNT",
        endAmount:123.55
    })
})

test('should NOT call dispatch endAmount is NOT valid',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('input').at(2).simulate('change',{target:{value:"123.555"}})
    expect(dispatchSpy).toHaveBeenCalledTimes(0)
})

test('should call dispatch with the correct action object when dates are changed',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('withStyles(DateRangePicker)').prop('onDatesChange')({startDate:moment(1000),endDate:moment(2000)})
    // expect(dispatchSpy).toHaveBeenNthCalledWith(1,{
    //     type:"SET_START_DATE",
    //     startDate:moment(1000).startOf('day').valueOf()
    // })
    // expect(dispatchSpy).toHaveBeenNthCalledWith(2,{
    //     type:"SET_END_DATE",
    //     endDate:moment(2000).endOf('day').valueOf()
    // })
    expect(dispatchSpy).toHaveBeenCalledTimes(2);
    expect(dispatchSpy.mock.calls[0]).toEqual([{
        type:"SET_START_DATE",
        startDate:moment(1000).startOf('day').valueOf()
    }])
    expect(dispatchSpy.mock.calls[1]).toEqual([{
        type:"SET_END_DATE",
        endDate:moment(2000).endOf('day').valueOf()
    }])
})

test('should update state when focus changes',()=>{
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<ExpenseFilters dispatch={dispatchSpy} filters={filters} />)
    wrapper.find('withStyles(DateRangePicker)').prop('onFocusChange')('startDate')
    expect(wrapper.state('focusedInput')).toBe('startDate')
})


