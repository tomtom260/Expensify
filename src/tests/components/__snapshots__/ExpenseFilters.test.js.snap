// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseFilters correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <select
    onChange={[Function]}
    value="DATE"
  >
    <option
      value="DATE"
    >
      Date
    </option>
    <option
      value="AMOUNT"
    >
      Amount
    </option>
    <option
      value="ALPHABETICAL"
    >
      Alphabetical
    </option>
  </select>
  <withStyles(DateRangePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customArrowIcon={null}
    customCloseIcon={null}
    customInputIcon={null}
    dayPickerNavigationInlineStyles={null}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    endDate={null}
    endDateId="your_unique_end_date_id"
    endDatePlaceholderText="End Date"
    firstDayOfWeek={null}
    focusedInput={null}
    hideKeyboardShortcutsPanel={true}
    horizontalMargin={0}
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    minimumNights={1}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPosition="navPositionTop"
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDatesChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableEndDate": [Function],
        "chooseAvailableStartDate": [Function],
        "clearDates": "Clear Dates",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsSelectedAsEndDate": [Function],
        "dateIsSelectedAsStartDate": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "roleDescription": "datepicker",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonthElement={null}
    renderMonthText={null}
    renderNavNextButton={null}
    renderNavPrevButton={null}
    renderWeekHeaderElement={null}
    reopenPickerOnClearDates={false}
    required={false}
    screenReaderInputMessage=""
    showClearDates={true}
    showDefaultInputIcon={false}
    small={false}
    startDate={null}
    startDateId="your_unique_start_date_id"
    startDatePlaceholderText="Start Date"
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <input
    onChange={[Function]}
    placeholder="Start Amount"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="End Amount"
    type="text"
    value=""
  />
</div>
`;

exports[`should render ExpenseFilters correctly with alt data 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value="bill"
  />
  <select
    onChange={[Function]}
    value="AMOUNT"
  >
    <option
      value="DATE"
    >
      Date
    </option>
    <option
      value="AMOUNT"
    >
      Amount
    </option>
    <option
      value="ALPHABETICAL"
    >
      Alphabetical
    </option>
  </select>
  <withStyles(DateRangePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customArrowIcon={null}
    customCloseIcon={null}
    customInputIcon={null}
    dayPickerNavigationInlineStyles={null}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    endDate={"1970-01-31T20:59:59.999Z"}
    endDateId="your_unique_end_date_id"
    endDatePlaceholderText="End Date"
    firstDayOfWeek={null}
    focusedInput={null}
    hideKeyboardShortcutsPanel={true}
    horizontalMargin={0}
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    minimumNights={1}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPosition="navPositionTop"
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDatesChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableEndDate": [Function],
        "chooseAvailableStartDate": [Function],
        "clearDates": "Clear Dates",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsSelectedAsEndDate": [Function],
        "dateIsSelectedAsStartDate": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "roleDescription": "datepicker",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonthElement={null}
    renderMonthText={null}
    renderNavNextButton={null}
    renderNavPrevButton={null}
    renderWeekHeaderElement={null}
    reopenPickerOnClearDates={false}
    required={false}
    screenReaderInputMessage=""
    showClearDates={true}
    showDefaultInputIcon={false}
    small={false}
    startDate={"1969-12-31T21:00:00.000Z"}
    startDateId="your_unique_start_date_id"
    startDatePlaceholderText="Start Date"
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <input
    onChange={[Function]}
    placeholder="Start Amount"
    type="text"
    value={150}
  />
  <input
    onChange={[Function]}
    placeholder="End Amount"
    type="text"
    value={1000}
  />
</div>
`;
